### PR TITLE
refactor(datasets): use revisions field on DatasetExample type

### LINKED ISSUE #3331: [datasets][gql] add revisions resolver to DatasetExample

### PR DESCRIPTION
**Pull request by axiomofjoy**
- Adds `RevisionKind` enum type.
- Adds `DatasetExampleRevision` type.
- Adds a `revisions` field to the `DatasetExample` type.
- Updates behavior of `DatasetExample` node interface to pull data from latest non-delete revision.

resolves #3331

### CODE REVIEW

File: app/src/pages/dataset/DatasetExamplesTable.tsx [hunk 1]
```diff
@@ -57,12 +59,13 @@ export function DatasetExamplesTable({
...
+        const revision = example.revision;
         return {
-          id,
-          input: JSON.stringify(input),
-          output: JSON.stringify(output),
-          metadata: JSON.stringify(metadata),
+          id: example.id,
+          input: revision.input,
+          output: revision.output,
+          metadata: revision.metadata,
```
Comment (axiomofjoy) [inline app/src/pages/dataset/DatasetExamplesTable.tsx:68]: Fixes display issue with fixtures.

File: app/src/pages/trace/SpanToDatasetExampleDialog.tsx [hunk 2]
```diff
@@ -121,9 +121,9 @@ export function SpanToDatasetExampleDialog({
             datasetId: newExample.datasetId,
             examples: [
               {
-                input: newExample.input,
-                output: newExample.output,
-                metadata: newExample.metadata,
+                input: JSON.parse(newExample.input),
+                output: JSON.parse(newExample.output),
+                metadata: JSON.parse(newExample.metadata),
```
Comment (axiomofjoy) [inline app/src/pages/trace/SpanToDatasetExampleDialog.tsx:126]: Resolves a bug where strings rather JSON objects were being sent to the `addExampleToDataset` mutation.

Comment (axiomofjoy) [reply app/src/pages/trace/SpanToDatasetExampleDialog.tsx]: Move to https://github.com/Arize-ai/phoenix/pull/3345

File: app/package.json [hunk 3]
```diff
@@ -87,7 +87,7 @@
     "build:relay": "relay-compiler",
     "watch": "./esbuild.config.mjs dev",
     "test": "jest --config ./jest.config.js",
-    "dev": "pnpm run dev:server & pnpm run build:static && pnpm run watch",
+    "dev": "pnpm run build:static && pnpm run watch",
```
Comment (mikeldking) [inline app/package.json:90]: I think you need to revert this change. You can make one called `dev:ui` if you want to run the server separately?

File: app/src/pages/dataset/DatasetExamplePage.tsx [hunk 4]
```diff
@@ -31,21 +31,23 @@ export function DatasetExamplePage() {
...
         example: node(id: $exampleId) {
           ... on DatasetExample {
-            input
-            output
-            metadata
+            revision {
+              input
+              output
+              metadata
+            }
```
Comment (mikeldking) [inline app/src/pages/dataset/DatasetExamplePage.tsx:38]:
```suggestion
            latestRevision: revision {
              input
              output
              metadata
            }
```

might make the most sense to alias here

File: src/phoenix/server/api/queries.py [hunk 5]
```diff
@@ -186,28 +187,44 @@ async def node(self, id: GlobalID, info: Info[Context, None]) -> Node:
...
+            revision_rowid = (
+                select(func.max(models.DatasetExampleRevision.id))
+                .where(models.DatasetExampleRevision.dataset_example_id == example_rowid)
+                .scalar_subquery()
+            )
             async with info.context.db() as session:
                 example_result = (
                     await session.execute(
-                        select(models.DatasetExampleRevision, models.DatasetExample)
+                        select(models.DatasetExample, models.DatasetExampleRevision)
```
Comment (mikeldking) [inline src/phoenix/server/api/queries.py:199]: Why not put this in a child resolver and use a data loader? This sorta falls under a bit of over-fetching (though I agree it's not really...)

Comment (axiomofjoy) [reply src/phoenix/server/api/queries.py:199]: Do you mean over-fetching because we grab the entire `DatasetExampleRevision`? I agree it's over-fetching if we don't need the `input`, `output`, or `metadata` if the user doesn't ask for the `revision`. We have to the do the join on the `dataset_example_revisions` table no matter what because we need to check that the example has not been deleted.

File: app/package.json (hunk 3 above)
Comment (axiomofjoy) [reply app/package.json]: addressed in c1f38bba3ed38153b8eed3f9725c2fd57f15584d

File: app/src/pages/dataset/DatasetExamplePage.tsx (hunk 4 above)
Comment (axiomofjoy) [reply app/src/pages/dataset/DatasetExamplePage.tsx]: addressed in fd82e0840a17decf47d7a43a0ee05dcacf801bcb

**Review by mikeldking — APPROVED** [review]
